### PR TITLE
Add code checkers

### PR DESCRIPTION
**Pull request by ricardoperez**
Add code check and refactored code to appease reek and rubocop.

- Add rspec test and romove minitests
- Fix reek and rubocop

### CODE REVIEW

File: lib/bing-ads-reporting/downloader.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    def initialize(report_url)
+      @report_url = report_url
+    end
+
+    def self.fetch_report(report_url)
+      new(report_url).fetch_report
+    end
+
+    def fetch_report
+      # logger.debug "Downloading Bing report from: #{report_url}"
```
Comment (robertino) [inline lib/bing-ads-reporting/downloader.rb]: any reason for this being here commented out?

File: lib/bing-ads-reporting/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module BingAdsReporting
-  VERSION = "3.0.0"
+  VERSION = '3.0.0'.freeze
```
Comment (robertino) [inline lib/bing-ads-reporting/version.rb]: we'll have to change the version for sure

**Review by robertino — APPROVED** [review]

File: lib/bing-ads-reporting/downloader.rb (hunk 1 above)
Comment (ricardoperez) [reply lib/bing-ads-reporting/downloader.rb]: good catch. I'll remove the comment.

File: lib/bing-ads-reporting/version.rb (hunk 2 above)
Comment (ricardoperez) [reply lib/bing-ads-reporting/version.rb]: done.

File: lib/bing-ads-reporting/downloader.rb (hunk 1 above)
Comment (ricardoperez) [reply lib/bing-ads-reporting/downloader.rb]: done.

**Review by chiting — APPROVED** [review]

**Pull request merged**